include/range/v3/view/indices.hpp

```diff
@@ -0,0 +1,123 @@
+/// \file
+// Range v3 library
+//
+//  Copyright Eric Niebler 2013-2014
+//  Copyright Gonzalo Brito Gadeschi
+//
+//  Use, modification and distribution is subject to the
+//  Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at
+//  http://www.boost.org/LICENSE_1_0.txt)
+//
+// Project home: https://github.com/ericniebler/range-v3
+//
+
+#ifndef RANGES_V3_VIEW_INDICES_HPP
+#define RANGES_V3_VIEW_INDICES_HPP
+
+#include <meta/meta.hpp>
+#include <range/v3/detail/satisfy_boost_range.hpp>
+#include <range/v3/range_fwd.hpp>
+#include <range/v3/range_concepts.hpp>
+#include <range/v3/utility/static_const.hpp>
+#include <range/v3/view/take_exactly.hpp>
+#include <range/v3/view/iota.hpp>
+
+namespace ranges
+{
+    inline namespace v3
+    {
+
+        namespace view
+        {
+            /// Half-open range of indices: [from, to).
+            struct indices_fn
+              : iota_view<std::size_t>
+            {
+                indices_fn() = default;
+
+                template<typename Val>
+                meta::if_c<
```

Comment:
@ericniebler ?

I just took this from `view::ints`, and since I did not know _why_ they were constrained that way instead of with `CONCEPT_REQUIRES_` I decided not to change it, but I see no reason against using `CONCEPT_REQUIRES_` here.